camelot/model/type_and_status.py

```diff
@@ -307,6 +307,28 @@ def change_status(self, new_status,
                                   thru_date = end_of_times())
         session.flush()
 
+
+class CurrentStatus(Action):
+
+    name = 'current_status'
+    render_hint = RenderHint.STATUS_BUTTON
+    verbose_name = ''
+
+    def get_state(self, model_context):
+        state = super().get_state(model_context)
+        obj = model_context.get_object()
+        if obj is not None:
+            field_attributes = model_context.admin.get_field_attributes(
+                'current_status'
+            )
+            current_status = obj.current_status
```

Comment:
Moet er nergens een check of assertion komen op het feit dat het object een instance van WithStatus is?